pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -52,13 +52,15 @@ var (
 	// the keys have been sourced from https://cloud.google.com/compute/docs/gpus/
 	// the values have been sourced from https://github.com/googleapis/google-api-go-client/blob/main/compute/v1/compute-gen.go
 	supportedGpuTypes = map[string]string{
-		"nvidia-tesla-k80":  "NVIDIA_K80_GPUS",
-		"nvidia-tesla-p100": "NVIDIA_P100_GPUS",
-		"nvidia-tesla-v100": "NVIDIA_V100_GPUS",
-		"nvidia-tesla-a100": "NVIDIA_A100_GPUS",
-		"nvidia-tesla-p4":   "NVIDIA_P4_GPUS",
-		"nvidia-tesla-t4":   "NVIDIA_T4_GPUS",
-		"nvidia-a100-80gb":  "NVIDIA_A100_80GB_GPUS",
+		"nvidia-tesla-k80":      "NVIDIA_K80_GPUS",
+		"nvidia-tesla-p100":     "NVIDIA_P100_GPUS",
+		"nvidia-tesla-v100":     "NVIDIA_V100_GPUS",
+		"nvidia-tesla-a100":     "NVIDIA_A100_GPUS",
+		"nvidia-tesla-p4":       "NVIDIA_P4_GPUS",
+		"nvidia-tesla-t4":       "NVIDIA_T4_GPUS",
+		"nvidia-a100-80gb":      "NVIDIA_A100_80GB_GPUS",
+		"nvidia-h100-mega-80gb": "NVIDIA_H100_GPUS",
```

Comment:
How about "NVIDIA_H100_MEGA_GPUS"?

Reply:
I don't think we have a choice here, if you read the comment above, this is mapping the keys from the documentation to the values in the SDK

Reply:
we could put `NVIDIA_H100_MEGA_GPUS` here, but since it isn't used in the AWS SDK, it would never match. unfortunately we are limited by what this version of the SDK will allow us to detect from the calls to AWS.